src/XrdCrypto/XrdCryptosslgsiAux.cc

```diff
@@ -900,9 +950,45 @@ int XrdCryptosslX509SignProxyReq(XrdCryptoX509 *xcpi, XrdCryptoRSA *kcpi,
       PRINT("inconsistent key loaded");
       return -kErrPX_BadEECkey;
    }
+
+   // These will be assigned dynamically allocated ssl structures later.
+   // An RAII style guard is used to free them on exit from this method.
+   EVP_PKEY       *ekpi = 0;
+   X509           *xpo  = 0;
+   X509_EXTENSION *ext  = 0;
+   PROXY_CERT_INFO_EXTENSION *pci   = 0;
+   STACK_OF(X509_EXTENSION)  *xrisk = 0;
+
+   class sslGuard {
```

Comment:
This definition of sslGaurd is almost identical to the previous version. It should be trivial to define a superset of sslGaurd and only have one definition placed at file level and used where need.